### PR TITLE
Feat/reclaim point of no return

### PR DESCRIPTION
**Pull request by JeremyIden**
Enhancement for issue #644. Adds a new checkbox to find the point of no return (last step with durability > 0 and enough CP to cast Reclaim), and split long macros at that point.

### CODE REVIEW

**Comment by Supamiu** [comment]
Hello 😄 

Thanks for your pull request, this is a nice first contribution.

I'm going to fix the conflicts and make a small review of your code to make sure everything is alright and poke the author of the issue to make sure he likes this solution 👍

File: apps/client/src/app/pages/simulator/components/macro-popup/macro-popup.component.ts [hunk 1]
```diff
@@ -40,10 +45,13 @@ export class MacroPopupComponent implements OnInit {
     this.macro = [[]];
     this.aactionsMacro = ['/aaction clear'];
     let totalLength = 0;
+    const reclaimBreakpoint = this.simulation.clone().run(true).simulation.lastPossibleReclaimStep;
     this.rotation.forEach((action) => {
+      console.log(reclaimBreakpoint);
```
Comment (Supamiu) [inline apps/client/src/app/pages/simulator/components/macro-popup/macro-popup.component.ts]: You left a console.log here ;)

File: apps/client/src/assets/i18n/de-DE.json [hunk 2]
```diff
@@ -501,6 +501,7 @@
   "SIMULATOR": {
     "Fixed_notification_number": "Konstanter Klangeffekt",
     "Starting_echo_number": "Klangeffekt #",
+    "Break_on_reclaim": "Split macros on Reclaim point-of-no-return",
```
Comment (Supamiu) [inline apps/client/src/assets/i18n/de-DE.json]: You don't have to put english names as placeholders, the translation library uses english as fallback no matter what :) You can probably leave these here for now tho, I'm curious how the translation tool will handle this (crowdin).

File: apps/client/src/app/pages/simulator/components/macro-popup/macro-popup.component.ts (hunk 1 above)
Comment (JeremyIden) [reply apps/client/src/app/pages/simulator/components/macro-popup/macro-popup.component.ts]: Ah, that was just to vet the code review process, making sure it's thorough! Like the programming version of a secret shopper.

Fixed, and merged in staging, thanks!

**Comment by Supamiu** [comment]
@JeremyIden Thanks for your contribution 😄 If you want to experiment with Teamcraft and add more stuff to it, feel free to poke me on Discord, I can also try to label more issues as newbie-friendly, even if it's hard for now with v5 ^^.

**Comment by JeremyIden** [comment]
Thank you, for the review and the invite! I'll have to do that once I get a
Discord, and make some more improvements when 5.0 is done.

Thanks for being welcoming!

**Comment by Supamiu** [comment]
@JeremyIden I just checked and having the english strings in the non english files is a problem because they don't get detected as missing by crowdin for translations. I'll fix that but for your next PRs, just pur english string in the en.json file 😄 Crowdin will handle the translations for the other files, don't worry

**Comment by mihawk90** [comment]
Heya, thanks for the works on this.
So I could just test this feature the first time and noticed some things off.

First off here's a rotation I was using:
```
/ac "Muscle Memory" <wait.3>
/ac "Inner Quiet" <wait.2>
/ac "Steady Hand II" <wait.2>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Prudent Touch" <wait.3>
/ac "Steady Hand II" <wait.2>
/ac "Prudent Touch" <wait.3>
/ac Innovation <wait.2>
/ac "Great Strides" <wait.2>
/ac "Byregot's Blessing" <wait.3>
/ac Observe <wait.3>
/ac "Focused Synthesis" <wait.3>
/ac Observe <wait.3>
/ac "Focused Synthesis" <wait.3>
/ac "Careful Synthesis III" <wait.3>
```

I noticed 2 things with the splits with different CP numbers.

First, with 180 CP (forgot to change them):
image

It seems every following chunk after the Reclaim split is using the same maximum amount of actions per chunk. I *think* you could use a reset in [macro-popup.component.ts Line 72](https://github.com/JeremyIden/ffxiv-teamcraft/blob/b5985ac1b63f0d56083d88747c77461ba967fa9e/apps/client/src/app/pages/simulator/components/macro-popup/macro-popup.component.ts#L72).

Second, the counter seems to be off by one:

image
Going through snapshot mode the breakpoint would actually be *after* Great Strides (Reclaim is 55):
image

The same for the 180 CP one, you can actually use one more Prudent Touch.

Another example using no spec this time, but Matcha HQ:
image

The breakpoint would be *after* Byregots.

And as a last example Spec + Matcha HQ:
image

This would actually split at the red line.

---

Also, 2 additional requests:
1. it would be really nice if the checkbox were persistent
2. Altering the chat message when the chunk ends due to Reclaim

**Pull request merged**